checkov/terraform/graph_builder/local_graph.py

```diff
@@ -418,7 +418,14 @@ def _find_vertex_index_relative_to_path(
         if self.use_new_tf_parser and relative_module_idx is None:
             module_dependency_by_name_key = source_module_object
         elif relative_module_idx is not None:
-            module_dependency_by_name_key = next(k for k, v in self.vertices_by_module_dependency.items() if v.get(BlockType.MODULE, []).__contains__(relative_module_idx))
+            relative_vertex = self.vertices[relative_module_idx]
+            relative_source_module_set = relative_vertex.source_module
+            if len(relative_source_module_set) == 0:
+                module_dependency_by_name_key = (None, None)
+            else:
+                relative_source_module_idx = list(relative_source_module_set)[0]
```

Comment:
```suggestion
                relative_source_module_idx = next(iter((relative_source_module_set))
```
it could be to use this, then you don't create a list to grab the first item

Reply:
so actually as we check the len before I think it's not necessary to add try/except